eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -91,6 +91,11 @@ stages:
                 deploy:
                   steps:
                     - checkout: self
+                    - template: /eng/common/pipelines/templates/steps/verify-changelog.yml
```

Comment:
This should be a step under VerifyReleaseVersion job, where the other verifications are performed, instead of under TagRepository.